mergify_engine/tests/functional/test_github_client.py

```diff
@@ -0,0 +1,65 @@
+# -*- encoding: utf-8 -*-
+#
+# Copyright © 2020 Mehdi Abaakouk <sileht@mergify.io>
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+
+import httpx
+import yaml
+
+from mergify_engine.clients import github
+from mergify_engine.tests.functional import base
+
+
+class TestGithubClient(base.FunctionalTestBase):
+    def test_github_client(self):
+        rules = {
+            "pull_request_rules": [
+                {
+                    "name": "simulator",
+                    "conditions": ["base!=master"],
+                    "actions": {"merge": {}},
+                }
+            ]
+        }
+        self.setup_repo(yaml.dump(rules), test_branches=["other"])
+        self.create_pr()
+        self.create_pr()
+        self.create_pr(base="other")
+
+        client = github.get_client(self.o_integration.login, self.r_o_integration.name)
+
+        pulls = list(client.items("pulls"))
+        self.assertEqual(2, len(pulls))
```

Comment:
maybe switch to `assert`?
I mean at some point we might want to stop using `unittest` no?

Reply:
This is even worse, we use testtools. I will switch the whole thing later.